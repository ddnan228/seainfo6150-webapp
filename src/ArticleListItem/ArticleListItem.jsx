import React from "react";
import HTMLText from "../HTMLText/HTMLText";

const ArticleListItem = (props) => {
  return (
    <article>
        <h1>{props.article.title}</h1>
        <p>{props.article.displayDate} {props.article.pubYear}<br /></p>
        <p>{props.article.shortText}</p>
    </article>
  );
};

export default ArticleListItem;
