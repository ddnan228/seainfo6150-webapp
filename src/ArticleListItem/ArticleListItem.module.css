.articleTitle {
  font-size: 20px;
  font-weight: bolder;
  color: darkred;
}

.articleBody {
  border: 2px solid #701717;
  border-radius: 10px;
  padding: 0px 5px 5px 5px;
  background: linear-gradient(to right, #ff5656, white 20%);
  box-shadow: 3px 3px 5px;

  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.time {
  font-weight: bold;
}
