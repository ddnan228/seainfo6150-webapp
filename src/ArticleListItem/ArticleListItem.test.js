import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem test", () => {

  it("renders correctly", () => {
    const test =
      {
        title: "The Statue of Liberty's torch heads to new museum",
        author: "Kate Farley",
        displayDate: "November 22nd 2018, 7:12 am",
        shortText: "The Statue of Liberty's original torch is getting a new home",
      }

    const { container } = render(<ArticleListItem article={test} />);
    expect(container).toMatchSnapshot();
  });
});
