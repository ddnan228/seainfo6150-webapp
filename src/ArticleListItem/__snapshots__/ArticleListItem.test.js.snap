// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem test renders correctly 1`] = `
<div>
  <div
    class="articleBody"
  >
    <h1
      class="articleTitle"
    >
      <u>
        The Statue of Liberty's torch heads to new museum
      </u>
    </h1>
    
    <button
      class="itemButton"
    >
      Show more
    </button>
  </div>
</div>
`;
